microservices/submission-service/src/replay-validation/replay-validation.service.spec.ts

```diff
@@ -0,0 +1,747 @@
+                    [2, 1],
+                ],
+            ];
+            const expectedOutput = [
+                [
+                    [1, 2],
+                    [3, 4],
+                ],
+                [
+                    [1, 2],
+                    [3, 4],
+                ],
+                [
+                    [1, 2],
+                    [3, 4],
+                ],
+            ];
+            expect(sortIds(inputGames)).toStrictEqual(expectedOutput);
+        });
+    });
+
+    describe("Check parent method branches properly:", () => {
+        it("Should validate as a scrim", async () => {
+            const scrimResponseObject = {
+                valid: true,
+                errors: [
+                    {
+                        error: "This is a scrim",
+                    },
+                ],
+            };
+            service.validateScrimSubmission = async _ => {
+                return scrimResponseObject;
+            };
+            expect(await service.validate(testSubmission)).toStrictEqual(scrimResponseObject);
+        });
+
+        it("Should validate as a match", async () => {
+            const matchResponseObject = {
+                valid: true,
+                errors: [
+                    {
+                        error: "This is a match",
+                    },
+                ],
+            };
+            service.validateMatchSubmission = async _ => {
+                return matchResponseObject;
+            };
+
+            expect(await service.validate(testMatchSubmission)).toStrictEqual(matchResponseObject);
+        });
+    });
+
+    describe("Validate Scrims:", () => {
+        it("Should validate that the two arrays are the same length", () => {
+            expect(service.validateNumberOfGames(testSubmission, testScrim)).toStrictEqual({valid: true});
+        });
+
+        it("Should pass validation because there are no error progress messages yet", () => {
+            expect(service.checkForProcessingErrors(testSubmission, testScrim)).toStrictEqual({valid: true});
+        });
+
+        it("Should return an error for different length arrays", () => {
+            testSubmission.items.push(testItem);
+            expect(service.validateNumberOfGames(testSubmission, testScrim)).toStrictEqual({
+                valid: false,
+                errors: [
+                    {
+                        error: "Incorrect number of replays submitted, expected 0 but found 1",
+                    },
+                ],
+            });
+        });
+
+        it("Should fail validation because there's an error progress message.", () => {
+            expect(service.checkForProcessingErrors(testSubmission, testScrim)).toStrictEqual({
+                valid: false,
+                errors: [
+                    {
+                        error: "Error encountered while parsing file ",
+                    },
+                ],
+            });
+        });
```

Comment:
These tests are not atomic; rearranging or adding ones in between would change the results

Reply:
Resolved in latest commit.